addons/predicates/tkr.go

```diff
@@ -18,3 +23,43 @@ func TKR(log logr.Logger) predicate.Funcs {
 		GenericFunc: func(e event.GenericEvent) bool { return true },
 	}
 }
+
+// ClusterHasLabel checks if the cluster has the given label
+func ClusterHasLabel(label string, logger logr.Logger) predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return processIfClusterHasLabel(label, e.ObjectNew, logger.WithValues("predicate", "updateEvent"))
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return processIfClusterHasLabel(label, e.Object, logger.WithValues("predicate", "createEvent"))
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return processIfClusterHasLabel(label, e.Object, logger.WithValues("predicate", "deleteEvent"))
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return processIfClusterHasLabel(label, e.Object, logger.WithValues("predicate", "genericEvent"))
+		},
+	}
+}
+
+// processIfClusterHasLabel determines if the input object is a cluster with a non-empty
+// value for the specified label. For other input object types, it returns true
+func processIfClusterHasLabel(label string, obj client.Object, logger logr.Logger) bool {
+	kind := obj.GetObjectKind().GroupVersionKind().Kind
+	clusterKind := reflect.TypeOf(clusterapiv1beta1.Cluster{}).Name()
```

Comment:
nit: There's no need to calculate `clusterKind` this every time `processIfClusterHasLabel()` is called. It's basically a constant, so we can take it out of the func body into a standalone var:
```go
var clusterKind = reflect.TypeOf(clusterapiv1beta1.Cluster{}).Name()
```